Realm/Tests/RealmTests.mm

```diff
@@ -2937,5 +2937,90 @@ - (void)testDeleteOpenRealmFile {
+        logs = newLogs;
+    }];
+    [RLMLogger setDefaultLogger:logger];
+
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertEqual([RLMLogger defaultLogger].level, RLMLogLevelAll);
+    XCTAssertTrue([logs rangeOfString:@"4 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Info
+    XCTAssertTrue([logs rangeOfString:@"7 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Trace
+
+    logs = @"";
+    [RLMLogger defaultLogger].level = RLMLogLevelInfo;
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertEqual([RLMLogger defaultLogger].level, RLMLogLevelInfo);
+    XCTAssertTrue([logs rangeOfString:@"4 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Info
+    XCTAssertTrue([logs rangeOfString:@"7 DB:" options:NSRegularExpressionSearch].location == NSNotFound); // Trace
+}
+
+- (void)testDefaultLogger {
+    __block NSString *logs = @"";
+    RLMLogger *logger = [[RLMLogger alloc] initWithLevel:RLMLogLevelOff logFunction:^(RLMLogLevel level, NSString * _Nonnull message) {
+        NSString *newLogs = [logs stringByAppendingFormat:@" %@ %lu %@", [NSDate now], level, message];
+        logs = newLogs;
+    }];
+    [RLMLogger setDefaultLogger:logger];
+    XCTAssertEqual([RLMLogger defaultLogger].level, RLMLogLevelOff);
 
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertTrue([logs length] == 0);
+
+    // Test LogLevel Info
+    logger.level = RLMLogLevelInfo;
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertTrue([logs length] >= 0);
+    XCTAssertTrue([logs rangeOfString:@"4 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Info
+    XCTAssertTrue([logs rangeOfString:@"7 DB:" options:NSRegularExpressionSearch].location == NSNotFound); // Trace
+
+    // Test LogLevel All
+    logger.level = RLMLogLevelAll;
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertTrue([logs length] >= 0);
+    XCTAssertTrue([logs rangeOfString:@"4 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Info
+    XCTAssertTrue([logs rangeOfString:@"7 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Trace
+
+    logs = @"";
+    // Init Custom Logger
+    [RLMLogger setDefaultLogger:[[RLMLogger alloc] initWithLevel:RLMLogLevelDebug logFunction:^(RLMLogLevel level, NSString * _Nonnull message) {
+        NSString *newLogs = [logs stringByAppendingFormat:@" %@ %lu %@", [NSDate now], level, message];
+        logs = newLogs;
+    }]];
+
+    XCTAssertEqual([RLMLogger defaultLogger].level, RLMLogLevelDebug);
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertTrue([logs rangeOfString:@"4 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Info
+    XCTAssertTrue([logs rangeOfString:@"7 DB:" options:NSRegularExpressionSearch].location == NSNotFound); // Trace
+}
+
+- (void)testCustomLoggerLogMessage {
```

Comment:
This test is testing a non-public function which appears to only be used in this test.

Reply:
This is used this API `- (void)logWithLevel:(RLMLogLevel)logLevel message:(NSString *)message, ... NS_SWIFT_UNAVAILABLE("");` which can be mistaken for this one `- (void)logLevel:(RLMLogLevel)logLevel message:(NSString *)message` which is explicitly exposed for the Swift API

Reply:
Why is it exposed for the Swift API? We're exposing a way for users to define a sink for our log messages, not for them to do their own logging.

Reply:
Changed to be used internally in case we want log things at SDK level